crates/bevy_widget/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+use bevy_app::{App, CoreStage, Plugin};
+use bevy_ecs::schedule::IntoSystemDescriptor;
+use bevy_ui::UiSystem;
+use progress_bar::update_progress_bars;
+
+mod progress_bar;
+
+pub use progress_bar::*;
+
+/// The basic plugin for Bevy Widget
+#[derive(Default)]
+pub struct WidgetPlugin;
+
+impl Plugin for WidgetPlugin {
+    fn build(&self, app: &mut App) {
+        app.register_type::<ProgressBarWidget>()
```

Comment:
I'm not sure about `ProgressBar` as a name. The core widget here seems to be much more general than just a bar for progress.

A rough idea for a more general/useful name:  `ResourceMeter`? Not sold on that idea, but it is more general?

Reply:
In general UI terminology, the name of the widget is usually "ProgressBar". 

- https://material.angular.io/components/progress-bar/overview
- https://react-bootstrap.github.io/components/progress/
- https://docs.unity3d.com/ScriptReference/UIElements.ProgressBar.html
- https://docs.unrealengine.com/4.27/en-US/API/Runtime/UMG/Components/UProgressBar/
- https://docs.godotengine.org/en/stable/classes/class_progressbar.html

A loading screen, a health bar, mana bar, XP bar, various resource trackers you find in games are just specific implementations of that widget.